src/routes/[network]/(account)/ram/buy/bytes/+page.svelte

```
@@ -1,62 +1,57 @@
 <script lang="ts">
 	import { getContext } from 'svelte';
-	import { Asset } from '@wharfkit/antelope';
+	import { Checksum256 } from '@wharfkit/antelope';
 
 	import { getSetting } from '$lib/state/settings.svelte.js';
 	import type { UnicoveContext } from '$lib/state/client.svelte';
 
 	import Button from '$lib/components/button/button.svelte';
 	import Code from '$lib/components/code.svelte';
-	import Input from '$lib/components/input/textinput.svelte';
 	import Label from '$lib/components/input/label.svelte';
 	import Stack from '$lib/components/layout/stack.svelte';
+	import Transaction from '$lib/components/transaction.svelte';
+	import NumberInput from '$lib/components/input/number.svelte';
+	import Cluster from '$lib/components/layout/cluster.svelte';
 
-	import { BuyRAMState } from './state.svelte.js';
+	import { BuyRAMState } from '../state.svelte.js';
 
 	const context = getContext<UnicoveContext>('state');
 	const { data } = $props();
 	const debugMode = getSetting('debug-mode', false);
 
 	const buyRamState: BuyRAMState = $state(new BuyRAMState(data.network.chain));
 
+	buyRamState.format = 'units';
+
+	let transactionId: Checksum256 | undefined = $state();
+
 	async function handleBuyRAM() {
 		if (!context.wharf || !context.wharf.session) {
 			alert('Not logged in');
 			return;
 		}
 
 		try {
-			await context.wharf.transact({
-				action: data.network.contracts.system.action('buyram', buyRamState.toJSON())
+			const transactionResult = await context.wharf.transact({
+				action: data.network.contracts.system.action('buyrambytes', buyRamState.toJSON())
 			});
-			alert('RAM purchase successful');
+
+			transactionId = transactionResult?.resolved?.transaction.id;
```

If we don't get a transaction ID, should this proceed? 

Instead of these optional variables, should we throw an error here if the result or resolved transaction isn't set?

---

That's already happening in the `transact` method. I do need the optionals though because that method is returning a `TransactResult` object and that type marks `resolved` as optional.